src/FastTags.hs

```diff
@@ -616,9 +616,15 @@ functionTags constructors = go []
     go tags (Pos _ (Token _ "(") : Pos pos (Token _ name)
             : Pos _ (Token prefix ")") : Pos _ (Token _ "::") : rest) =
         (reverse $ mkTag pos prefix name opTag : tags, rest)
+    go tags (Pos _ (Token _ "(") : Pos pos (Token _ name)
+            : Pos _ (Token prefix ")") : Pos _ (Token _ "∷") : rest) =
+        (reverse $ mkTag pos prefix name opTag : tags, rest)
     go tags (Pos pos (Token prefix name) : Pos _ (Token _ "::") : rest)
         | functionName constructors name =
             (reverse $ mkTag pos prefix name funcTag : tags, rest)
+    go tags (Pos pos (Token prefix name) : Pos _ (Token _ "∷") : rest)
+        | functionName constructors name =
+            (reverse $ mkTag pos prefix name funcTag : tags, rest)
```

Comment:
I'd like to avoid the copy paste, could you put the two "::" variants into a list and put it in the guard, e.g.

| token `elem` typeOfTokens && functionName constructors name = ...